app/models/manageiq/providers/amazon/inventory/targets/event_payload_vm.rb

```diff
@@ -0,0 +1,25 @@
+class ManageIQ::Providers::Amazon::Inventory::Targets::EventPayloadVm < ManageIQ::Providers::Amazon::Inventory::Targets
+  def initialize_inventory_collections
+    instance_ems_ref = target[:full_data]["configurationItem"]["resourceId"]
```

Comment:
The ems_ref is for InventoryCollections init, so we know how to compare it. Then the data itself comes from event, in the instances method.

Which will result in create/update/delete based on the state of our db and the event